File: 0706-design-hashmap/0706-design-hashmap.py
class MyHashMap:

    def __init__(self):
        self.hashmap = []        

    def put(self, key: int, value: int) -> None:
        if key >= len(self.hashmap):
            self.hashmap.extend([-1]*(key+1-len(self.hashmap)))
        self.hashmap[key] = value

    def get(self, key: int) -> int:
        if key < len(self.hashmap):
            return self.hashmap[key]
        return -1
    def remove(self, key: int) -> None:
        if key < len(self.hashmap):
            self.hashmap[key] = -1


# Your MyHashMap object will be instantiated and called as such:
# obj = MyHashMap()
# obj.put(key,value)
# param_2 = obj.get(key)
# obj.remove(key)

File: 0706-design-hashmap/0706-design-hashmap.cpp
class MyHashMap {
private:
    vector<int> map;
public:
    MyHashMap() {}
    
    void put(int key, int value) {
        if(map.size() <= key){
            map.resize(key+1, -1);
        }
        map[key] = value;
    }
    
    int get(int key) {
        if(map.size() > key){
            return map[key];
        }
        return -1;
    }
    
    void remove(int key) {
        if(key < map.size()){
            map[key] = -1;
        }
    }
};

/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap* obj = new MyHashMap();
 * obj->put(key,value);
 * int param_2 = obj->get(key);
 * obj->remove(key);
 */